### DISCUSSION TITLE
Example script using RIFE?  (re: "vsmlrt.RIFE: only constant format 16/32 bit float input supported")

### DISCUSSION THREAD

**Discussion by sofakng** [AmusementClub/vs-mlrt#46] [General]
I'm trying to write a simple script to interpolate frames using RIFE.  I'm completely new to VapourSynth (and vs-mlrt) so I apologize for the basic question.

I'm using AviSynthFilter (v1.4.6), MPC-HC, and vsml-rt v13.

Here is my script:
```
import vapoursynth as vs
from vsmlrt import RIFE, RIFEModel, Backend
flt = RIFE(VpsFilterSource, model=RIFEModel.v4_4, backend=Backend.TRT()).set_output()
```

The error displayed is "vsmlrt.RIFE: only constant format 16/32 bit float input supported".  VapourSynth reports the clip is using integer sample type with 8 bits per sample so it sounds like I need to convert it to FP16/FP32 before passing to RIFE?

Thanks for any help.

**Comment by WolframRhodium** [comment]
Yes, the conversion could be done by `core.resize.Bicubic(yuv420, format=vs.RGBS, matrix_in_s="709")`.

**Comment by sofakng** [comment]
Thanks a bunch for the help.

I'll give this a try... I also noticed that fmtconv is used in parts of the vsmlrt scripts so I can give that a try.

Do you have any full example script to convert files in real-time or other tips?

**Comment by WolframRhodium** [comment]
```python3
import vapoursynth as vs
from vapoursynth import core
import vsmlrt
from vsmlrt import RIFEModel, Backend

backend = Backend.TRT(fp16=True, num_streams=2, use_cuda_graph=True, output_format=1)

src = core.lsmas.LWLibavSource("source.mp4")
ph = (src.height + 31) // 32 * 32 - src.height
pw = (src.width  + 31) // 32 * 32 - src.width
padded = src.std.AddBorders(right=pw, bottom=ph).resize.Bicubic(format=vs.RGBH, matrix_in_s="709")
flt = vsmlrt.RIFE(padded, model=RIFEModel.v4_4, backend=backend)
res = flt.resize.Bicubic(format=vs.YUV420P8, matrix_s="709").std.Crop(right=pw, bottom=ph)
res.set_output()
```

**Reply by sofakng** [reply]
Thanks so much!
